From 73e94f0290e6125209f73bde57442ec7bb76f59e Mon Sep 17 00:00:00 2001
From: Maxime Chevallier <maxime.chevallier@bootlin.com>
Date: Thu, 6 Dec 2018 16:26:03 +0100
Subject: [PATCH 123/386] arm64: dts: marvell: add Cybertan board description

Change-Id: I26db1fb22d08fc2d70d29f49ef8143ed640531dc
Signed-off-by: Maxime Chevallier <maxime.chevallier@bootlin.com>
[Add bindings documentation, fix checkpatch]
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/6491
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt     |   3 +
 arch/arm64/boot/dts/marvell/Makefile          |   1 +
 .../boot/dts/marvell/armada-8040-ucpe.dts     | 429 ++++++++++++++++++
 3 files changed, 433 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 1857b2bb16e1..668ad08674fa 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -63,6 +63,9 @@ Boards:
    - "marvell,armada8040-mcbin-single-shot"
       when the board being used is the MACCHIATOBin single shot community board
 
+   - "marvell,armada8040-ucpe"
+      when the board being used is the Cybertan UCPE switch
+
 Example:
 
 compatible = "marvell,armada7040-db", "marvell,armada7040",
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 3e8ee68decbd..568bfdcbae35 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -21,6 +21,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-G.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-H.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-single-shot.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-ucpe.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-A.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts b/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts
new file mode 100644
index 000000000000..90c5fd9cf5d1
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts
@@ -0,0 +1,429 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for the Cybertan Armada 8040 based switch
+ */
+
+#include "armada-8040.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Cybertan 8040 uCPE Switch";
+	compatible = "marvell,armada8040-ucpe", "marvell,armada8040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp1_eth0;
+		ethernet2 = &cp1_eth1;
+	};
+
+	/* Regulator labels correspond with schematics */
+	v_3_3: regulator-3-3v {
+		compatible = "regulator-fixed";
+		regulator-name = "v_3_3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		status = "okay";
+	};
+
+	v_vddo_h: regulator-1-8v {
+		compatible = "regulator-fixed";
+		regulator-name = "v_vddo_h";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+		status = "okay";
+	};
+	reg_usb3h0_vbus: USB3-Right-Port-vbus {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&CP0_usb3_port0_en>;
+		regulator-name = "reg-usb3h0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-low;
+		gpio = <&cp0_gpio2 11 GPIO_ACTIVE_LOW>; /* GPIO[43]*/
+		status = "okay";
+	};
+	reg_usb3h1_vbus: USB3-LEFT-Port-vbus {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&CP0_usb3_port1_en>;
+		regulator-name = "reg-usb3h1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-low;
+		gpio = <&cp0_gpio2 10 GPIO_ACTIVE_LOW>; /*GPIO[44]*/
+		status = "okay";
+	};
+
+	usb3h0_phy: usb3_phy0 {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&reg_usb3h0_vbus>;
+	};
+
+	usb3h1_phy: usb3_phy1 {
+		compatible = "usb-nop-xceiv";
+		vcc-supply = <&reg_usb3h1_vbus>;
+	};
+
+	sfp_eth0: sfp-eth0 {
+		compatible = "sff,sfp";
+		i2c-bus = <&sfpp0_i2c>;
+		los-gpio = <&cp1_gpio1 8 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&cp1_gpio1 11 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&cp1_gpio1 10 GPIO_ACTIVE_HIGH>;
+		tx-fault-gpio = <&cp1_gpio1 9 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&cp1_pinctrl_cpio_sfp0>;
+		status = "okay";
+	};
+};
+
+&uart0 {
+	status = "okay";
+	pinctrl-0 = <&uart0_pins>;
+	pinctrl-names = "default";
+};
+
+&ap_sdhci0 {
+	bus-width = <8>;
+	/*
+	 * Not stable in HS modes - phy needs "more calibration", so add
+	 * the "slow-mode" and disable SDR104, SDR50 and DDR50 modes.
+	 */
+	marvell,xenon-phy-slow-mode;
+	no-1-8-v;
+	no-sd;
+	no-sdio;
+	non-removable;
+	status = "okay";
+	vqmmc-supply = <&v_vddo_h>;
+};
+
+&cp0_pinctrl {
+	CP0_usb3_port0_en: xhci0-vbus-pins {
+		marvell,pins = "mpp43";
+		marvell,function = "gpio";
+	};
+	CP0_usb3_port1_en: xhci1-vbus-pins {
+		marvell,pins = "mpp42";
+		marvell,function = "gpio";
+	};
+	cp0_ge_mdio_pins: ge-mdio-pins {
+		marvell,pins = "mpp40", "mpp41";
+		marvell,function = "ge";
+	};
+	cp0_i2c1_pins: i2c1-pins {
+		marvell,pins = "mpp35", "mpp36";
+		marvell,function = "i2c1";
+	};
+	cp0_i2c0_pins: i2c0-pins {
+		marvell,pins = "mpp37", "mpp38";
+		marvell,function = "i2c0";
+	};
+};
+
+&cp0_i2c0 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c0_pins>;
+	status = "okay";
+};
+
+&cp0_i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c1_pins>;
+	status = "okay";
+
+	i2c-switch@70 {
+		compatible = "nxp,pca9548";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x70>;
+
+		sfpp0_i2c: i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+		};
+		sfpp1_i2c: i2c@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+		};
+		sfp_1g_i2c: i2c@2 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <2>;
+		};
+	};
+};
+
+&cp0_crypto {
+	status = "okay";
+};
+
+&cp0_usb3_0 {
+	status = "okay";
+	usb-phy = <&usb3h0_phy>;
+	phys = <&cp0_comphy1 0 >;
+	phy-names = "usb";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+	usb-phy = <&usb3h1_phy>;
+	phys = <&cp0_comphy3 1 >;
+	phy-names = "usb";
+};
+
+&cp1_gpio1 {
+	status = "okay";
+	interrupt-controller;
+	#interrupt-cells = <2>;
+};
+
+&cp0_xmdio {
+	status = "okay";
+};
+
+&cp0_mdio {
+	pinctrl-names = "ge";
+	pinctrl-0 = <&cp0_ge_mdio_pins>;
+	status = "okay";
+
+	#address-cells = <1>;
+	#size-cells = <0>;
+	#interrupt-cells = <2>;
+
+	switch0: switch@0 {
+		compatible = "marvell,mv88e6190";
+		pinctrl-0 = <&cp1_pinctrl_gpio_switch0>;
+		pinctrl-names = "default";
+		reg = <0>;
+		dsa,member = <0 0>;
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		eeprom-length = <65536>;
+
+		interrupt-parent = <&cp1_gpio1>;
+		interrupts = <30 IRQ_TYPE_LEVEL_LOW>;
+		interrupt-controller;
+		#interrupt-cells = <2>;
+
+		reset-gpios = <&cp1_gpio1 31 GPIO_ACTIVE_LOW>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			/*
+			 * port9: port@9 {
+			 *     reg = <0x9>;
+			 *     label = "cpu";
+			 *     ethernet = <&cp1_eth0>;
+			 *     phy-mode = "rxaui";
+			 * };
+			 */
+			port10: port@10 {
+				reg = <0xa>;
+				label = "cpu";
+				ethernet = <&cp1_eth1>;
+				phy-mode = "2500base-x";
+				fixed-link {
+					speed = <2500>;
+					full-duplex;
+				};
+			};
+			/*
+			 * Interface labels correspond to the switch's front
+			 * panel numbering
+			 */
+			port@1 {
+				reg = <1>;
+				label = "lan1";
+				phy-handle = <&switch0port1>;
+			};
+
+			port@2 {
+				reg = <2>;
+				label = "lan2";
+				phy-handle = <&switch0port2>;
+			};
+
+			port@3 {
+				reg = <3>;
+				label = "lan3";
+				phy-handle = <&switch0port3>;
+			};
+			port@4 {
+				reg = <4>;
+				label = "lan5";
+				phy-handle = <&switch0port4>;
+			};
+
+			port@5 {
+				reg = <5>;
+				label = "lan6";
+				phy-handle = <&switch0port5>;
+			};
+
+			port@6 {
+				reg = <6>;
+				label = "lan7";
+				phy-handle = <&switch0port6>;
+			};
+
+			port@7 {
+				reg = <7>;
+				label = "lan8";
+				phy-handle = <&switch0port7>;
+			};
+
+			port@8 {
+				reg = <8>;
+				label = "lan4";
+				phy-handle = <&switch0port8>;
+			};
+		};
+
+		mdio-bus {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			switch0port1: switch0port@1 {
+				reg = <1>;
+				interrupt-parent = <&switch0>;
+				interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port2: switch0port@2 {
+				reg = <2>;
+				interrupt-parent = <&switch0>;
+				interrupts = <2 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port3: switch0port@3 {
+				reg = <3>;
+				interrupt-parent = <&switch0>;
+				interrupts = <3 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port4: switch0port@4 {
+				reg = <4>;
+				interrupt-parent = <&switch0>;
+				interrupts = <4 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port5: switch0port@5 {
+				reg = <5>;
+				interrupt-parent = <&switch0>;
+				interrupts = <5 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port6: switch0port@6 {
+				reg = <6>;
+				interrupt-parent = <&switch0>;
+				interrupts = <6 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port7: switch0port@7 {
+				reg = <7>;
+				interrupt-parent = <&switch0>;
+				interrupts = <7 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port8: switch0port@8 {
+				reg = <8>;
+				interrupt-parent = <&switch0>;
+				interrupts = <8 IRQ_TYPE_LEVEL_HIGH>;
+			};
+			switch0port10: switch0port@10 {
+				reg = <10>;
+				interrupt-parent = <&switch0>;
+				interrupts = <10 IRQ_TYPE_LEVEL_HIGH>;
+			};
+		};
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy2 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth0>;
+};
+
+&cp1_pinctrl {
+	cp1_pinctrl_gpio_switch0: pinctrl_gpio_switch0 {
+		marvell,pins = "mpp30", "mpp31";
+		marvell,function = "gpio";
+	};
+
+	cp1_pinctrl_cpio_sfp0: cp1_pinctrl_cpio_sfp0 {
+		marvell,pins = "mpp8", "mpp9", "mpp10", "mpp11";
+		marvell,function = "gpio";
+	};
+};
+
+&cp1_crypto {
+	status = "okay";
+};
+
+&cp1_sata0 {
+	status = "okay";
+	sata-port@0 {
+		status = "okay";
+		phys = <&cp1_comphy1 0>;
+	};
+	sata-port@1 {
+		status = "okay";
+		phys = <&cp1_comphy0 1>;
+	};
+};
+
+&cp1_usb3_0 {
+	status = "okay";
+	phys = <&cp1_comphy2 0 >;
+	phy-names = "usb";
+};
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+&cp1_eth0 {
+	status = "disabled";
+
+	phy-mode = "rxaui";
+	phys = <&cp1_comphy4 0>, <&cp1_comphy5 0>;
+	fixed-link {
+		speed = <10000>;
+		full-duplex;
+	};
+};
+
+&cp1_eth1 {
+	status = "okay";
+
+	phy-mode = "2500base-x";
+	phys = <&cp1_comphy3 1>;
+	fixed-link {
+		speed = <2500>;
+		full-duplex;
+	};
+};
-- 
2.17.1

